scheds/rust/scx_rustland/src/main.rs

```diff
@@ -351,11 +351,15 @@ impl<'a> Scheduler<'a> {
         sum_exec_runtime: u64,
         weight: u64,
         min_vruntime: u64,
+        max_slice_ns: u64,
     ) {
         // Add cputime delta normalized by weight to the vruntime (if delta > 0).
         if sum_exec_runtime > task_info.sum_exec_runtime {
-            let delta = (sum_exec_runtime - task_info.sum_exec_runtime) / weight;
-            task_info.vruntime += delta;
+            let mut delta = sum_exec_runtime - task_info.sum_exec_runtime;
+            // Never account more than max_slice_ns. This helps to prevent starving a task for too
+            // long in the scheduler task pool.
+            delta = delta.min(max_slice_ns);
```

Comment:
I'm curious why this is necessary. Maybe this is needed to mask a different issue such as failing to wind current vtime forward when it should be? Or is this necessary because sometimes the scheduler takes a bit too long to kick them off cpu and their execution pattern becomes too chunky?

Reply:
> I'm curious why this is necessary. Maybe this is needed to mask a different issue such as failing to wind current vtime forward when it should be? Or is this necessary because sometimes the scheduler takes a bit too long to kick them off cpu and their execution pattern becomes too chunky?

I think I was just trying to fix what was fixed by `2900b20 ("scx_rustland: evaluate the proper vruntime delta")` here, I should have squashed the commits together...

However, now I can see a case where the delta can be bigger than `slice_ns`, but that's a side effect of the fact that we don't clean entries synchronously from the `self.task_info` HashMap when the corresponding tasks exit.

If a pid is reused we would (incorrectly) get the previous `TaskInfo` entry with the `sum_exec_runtime` of the previous task.

Do we have a callback that we can use in `sched_ext_ops` to detect when a task exits in a synchronous way? I was trying to use `.disable`, but it seems to be called only when the scheduler exits, not when the task exits... is this the expected behavior? Am I missing something?

Reply:
`.disable` should be called when a task gets freed, which may take place a bit after the task exits.

Reply:
> `.disable` should be called when a task gets freed, which may take place a bit after the task exits.

ok, I'll try again with the `.disable` callback to notify the user-space sched when tasks are exiting, it's more elegant than running the periodic garbage collector.

(Maybe we still need to do the garbage collector in case we fail to notify the user-space scheduler, but I don't think it's a big problem for now).

Thanks!